src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -668,8 +668,16 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
       this.legend = layerParams['legend'] ? layerParams['legend'] : null;
 
       if (!this.evalscript) {
-        this.evalscript = layerParams['evalscript'] ? layerParams['evalscript'] : null;
+        const dataProductUrl = layerParams['dataProduct'];
+        this.evalscript = layerParams['evalscript'];
+        if (!this.evalscript && dataProductUrl) {
+          const response = await fetchDataProduct(dataProductUrl, innerReqConfig);
+          this.evalscript = response.data?.evalScript;
+          // now that evalscript here, dataProduct=null will make rest of logic use ApiType.PROCESSING
+          this.dataProduct = null;
```

Comment:
I think setting `this.dataProduct` to `null` is not needed, because `createProcessingPayload()` used in `getMap()` checks if evalscript is set first (and uses it (instead of `dataProduct`) if it is).

Reply:
createProcessingPayload is not called if apiType is determined wms.

Reply:
True, in that case the `getMap` from the parent class is called, but it does not use `dataProduct` so it's not important if the value is kept or not. Let's leave it the way you did it.